### PR TITLE
#533: Update requireAlignedObjectValues rule to work with whitespace aligned via spaces and/or tabs

### LINKED ISSUE #533: requireAlignedObjectValues does not work with tabs
Hello,

requireAlignedObjectValues rule does not work with tabs but only with spaces. Would it be possible to work with spaces too?

Example:
```javascript
q = {
	qwe    : 1,
	asdfgh : 2
};
```
Output:
```
$ jscs test.js 
No code style errors found.
```

---

And if tabs are used (eq to 4 spaces)
```javascript
q = {
	qwe		: 1,
	asdfgh	: 2
};
```
```
$ jscs test.js 
Alignment required at test.js :
     1 |q = {
     2 | qwe  : 1,
--------------^
     3 | asdfgh : 2
     4 |};
```

Also you can notice that there is no error on line 3 because there is only a single whitespace character and jscs thinks it is aligned fine.

---

Also this seems to be fine too using lots of tabs. I guess this is because there are the same number of tabs as spaces would be.
```javascript
q = {
	qwe				: 1,
	asdfgh	: 2
};
```

Thanks!

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/3187440-requirealignedobjectvalues-does-not-work-with-tabs?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by SgtPooki**
This fixes #533. 

I verified that this fix works in the following scenarios:

- Only tabs used for alignment
- Only spaces used for alignment
- tabs used for some object keys and spaces used for others
- tabs and spaces used for some object keys, spaces used for some, and tabs used for others.

It should work in all scenarios and give you an accurate reading as to whether your objects are aligned or not as long as you update the requireAlignedObjectValues rule to be set to something like the following:
```
{
  "requireAlignedObjectValues" : {
    "mode" : "any string that was valid before",
    "tabSize" : "the amount of whitespace that your tabs count as"
  }
```

### CODE REVIEW

**Comment by mikesherov** [comment]
@markelog reopened it so that when I abstract tabSize out of a specific ruleset and into config like dialect, I know which rules to modify, including this.

**Comment by markelog** [comment]
@mikesherov still relevant?

File: lib/rules/require-aligned-object-values.js [hunk 1]
```diff
@@ -12,10 +40,13 @@ module.exports.prototype = {
             'skipWithFunction': 'ignoreFunction',
             'skipWithLineBreak': 'ignoreLineBreak'
         };
+
```
Comment (zxqfox) [inline lib/rules/require-aligned-object-values.js]: @SgtPooki As a man who merging like thouthands line of code each 1-2 weeks I can answer you: it's just stupidly annoying when someone adding empty lines and another one removing them. Personally in my team we have non-verbal rules: 1. Don't add empty lines in places in the code unless you specially fixing code style; 2. Use separate issues and commits for code-style fixes; 3. Changes should be focused on issues.

I'm totally agree with @mikesherov that there are no reasons to toggle these empty lines in this issue.

**Comment by mikesherov** [comment]
@markelog you can reclose.

File: lib/rules/require-aligned-object-values.js (hunk 1 above)
Comment (SgtPooki) [inline lib/rules/require-aligned-object-values.js]: @zxqfox I totally agree with separating style commits from code changes. I did not realize that is what he was talking about, or that I had even done it. That is why I asked for clarification.

Comment (zxqfox) [inline lib/rules/require-aligned-object-values.js]: I'm checking what empty lines I've added with `git show`, `git show HEAD~2` or `git diff origin/master` (after rebasing of origin/master) and just removing random spaces I've accidentally added: to make commits focused on issues.

File: README.md [hunk 2]
```diff
@@ -1454,12 +1454,16 @@ var x = {
 
 Requires proper alignment in object literals.
 
-Type: `String`
+Type: `String` or `Object`
```
Comment (zxqfox) [inline README.md]: TypeS.

File: README.md [hunk 3]
```diff
@@ -1454,12 +1454,16 @@ var x = {
...
 
 Values:
- - `"all"` for strict mode,
- - `"ignoreFunction"` ignores objects if one of the property values is a function expression,
- - `"ignoreLineBreak"` ignores objects if there are line breaks between properties
+ - `String`:
```
Comment (zxqfox) [inline README.md]: Not sure about this change.

File: lib/rules/require-aligned-object-values.js [hunk 4]
```diff
@@ -1,10 +1,38 @@
...
+            this.configureMode(config.mode, true);
+            this.configureTabSize(config.tabSize);
+
+        } else {
+            this.configureMode(config);
+        }
+    },
+
+    configureTabSize: function(tabSize) {
```
Comment (zxqfox) [inline lib/rules/require-aligned-object-values.js]: :+1: for moving them as local functions into `configure`

File: lib/rules/require-aligned-object-values.js [hunk 5]
```diff
@@ -23,17 +50,33 @@ module.exports.prototype = {
...
+
+        return {
+            pos: colonPosition,
+            colon: colon
+        };
+    },
+
     check: function(file, errors) {
-        var tokens = file.getTokens();
+
```
Comment (zxqfox) [inline lib/rules/require-aligned-object-values.js]: Intentional empty line.

File: lib/rules/require-aligned-object-values.js [hunk 6]
```diff
@@ -23,17 +54,33 @@ module.exports.prototype = {
         return 'requireAlignedObjectValues';
     },
 
+    _getColonPosition: function(file, property) {
```
Comment (zxqfox) [inline lib/rules/require-aligned-object-values.js]: :+1: You using it in `check` and no reason to left it in prototype.

**Pull request closed without merging**